apps/omg_watcher_info/lib/omg_watcher_info/db/txoutput.ex

```diff
@@ -79,23 +82,23 @@ defmodule OMG.WatcherInfo.DB.TxOutput do
     )
   end
 
-  def get_utxos(owner) do
-    query =
-      from(
-        txoutput in __MODULE__,
-        preload: [:ethevents],
-        left_join: ethevent in assoc(txoutput, :ethevents),
-        # select txoutputs by owner that have neither been spent nor have a corresponding ethevents exit events
-        where: txoutput.owner == ^owner and is_nil(txoutput.spending_txhash) and (is_nil(ethevent) or fragment("
- NOT EXISTS (SELECT 1
-             FROM ethevents_txoutputs AS etfrag
-             JOIN ethevents AS efrag ON
-                      etfrag.root_chain_txhash_event=efrag.root_chain_txhash_event
-                      AND efrag.event_type IN (?)
-                      AND etfrag.child_chain_utxohash = ?)", "standard_exit", txoutput.child_chain_utxohash)),
-        order_by: [asc: :blknum, asc: :txindex, asc: :oindex]
-      )
+  @spec get_utxos(keyword) :: OMG.Utils.Paginator.t()
+  def get_utxos(params) do
+    address = Keyword.fetch!(params, :address)
+    paginator = Paginator.from_constraints(params, @default_get_utxos_limit)
+    %{limit: limit, page: page} = paginator.data_paging
+    offset = (page - 1) * limit
+
+    address
+    |> query_get_utxos()
+    |> from(limit: ^limit, offset: ^offset)
+    |> Repo.all()
+    |> Paginator.set_data(paginator)
+  end
 
+  @spec get_all_utxos(OMG.Crypto.address_t()) :: list()
+  defp get_all_utxos(address) do
```

Comment:
mixing private and public functions, the rule is, public on top, private at the bottom.